docs/guides/how_to_guides/creating_batches/how_to_load_a_spark_dataframe_as_a_batch.rst

```diff
@@ -85,105 +85,174 @@ This guide will help you load a Spark DataFrame as a Batch for use in creating E
+          .. code-block:: python
 
-            .. code-block:: python
+            suite: ExpectationSuite = context.create_expectation_suite("insert_your_expectation_suite_name_here")
 
-                suite = context.create_expectation_suite("insert_your_expectation_suite_name_here")
+        2. Construct a Runtime Batch Request
 
-        3. Construct a BatchRequest
+          We will create a ``RuntimeBatchRequest`` and pass it our Spark DataFrame or path via the ``runtime_parameters`` argument, under either the ``batch_data`` or ``path`` key. The ``batch_identifiers`` argument is required and must be a non-empty dictionary containing all of the Batch Identifiers specified in your Runtime Data Connector configuration.
 
-            We will create a ``BatchRequest`` and pass it our DataFrame via the ``batch_data`` argument.
+          If you are providing a filesystem path instead of a materialized DataFrame, you may use either an absolute or relative path (with respect to the current working directory). Under the hood, Great Expectations will instantiate a Spark Dataframe using the appropriate ``spark.read.*`` method, which will be inferred from the file extension. If your file names do not have extensions, you can specify the appropriate reader method explicitly via the ``batch_spec_passthrough`` argument. Any Spark reader options (i.e. ``delimiter`` or ``header``) that are required to properly read your data can also be specified with the ``batch_spec_passthrough`` argument, in a dictionary nested under a key named ``reader_options``.
 
-            Attributes inside the ``data_connector_query`` are optional - you can use them to attach additional metadata to your DataFrame. When configuring the Data Connector, you used ``batch_identifiers`` to define which keys are allowed.
+          Example ``great_expectations.yml`` Datsource configuration:
 
-            .. code-block:: python
-
-                from great_expectations.core.batch import BatchRequest
-
-                batch_request = BatchRequest(
-                    datasource_name="insert_your_spark_datasource_name_here",
-                    data_connector_name="insert_your_runtime_data_connector_name_here",
-                    data_asset_name="insert_your_data_asset_name_here",
-                    batch_data=insert_your_dataframe_here,
-                    data_connector_query={
-                        "batch_identifiers": {
-                            "some_key_maybe_pipeline_stage": "ingestion step 1",
-                            "some_other_key_maybe_run_id": "run 18"
-                        }
-                    }
-                )
+          .. code-block:: yaml
 
+            my_spark_datasource:
+              execution_engine:
+                module_name: great_expectations.execution_engine
+                class_name: SparkExecutionEngine
+              module_name: great_expectations.datasource
```

Comment:
I believe this is `SparkDFExecutionEngine`

Reply:
Good catch. Fixed!